### ISSUE TITLE
Class 'message' has no 'split' member (no-member) when calling split on Exception.message

### ISSUE THREAD

**Issue by crbunney** [PyCQA/pylint#962]
PyLint generates a spurious no-member Error when interacting with `Exception.message`. It incorrectly identifies the type of `Exception.message` (which is a `str`) and incorrectly believes that `Exception.message` doesn't have a `split` method (which is does)

### Steps to reproduce
I created this test file:
```python
try:
   raise
except Exception as e:
   print type(e)
   print dir(e)
   print ''
   print type(e.message)
   print dir(e.message)
   e.message.split(' ')
``` 
I ran this command on it: `pylint -r n --unsafe-load-any-extension y --errors-only test.py`
This output was generated:
```
No config file found, using default configuration
************* Module test
E:  4, 3: Class 'message' has no 'split' member (no-member)
E:  4, 3: Class 'message' has no 'split' member (no-member)
```

### Current behavior
* The call to `e.message.split` is flagged as a potential missing member
* 2 Error messages are generated despite the call only being made once

### Expected behavior
No error raised

### pylint --version output
```
No config file found, using default configuration
pylint 1.5.6,
astroid 1.4.6
Python 2.7.11 (default, Jan 22 2016, 08:29:18)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

### pip freeze
```
astroid==1.4.6
colorama==0.3.7
lazy-object-proxy==1.2.2
pylint==1.5.6
six==1.10.0
wrapt==1.10.8
```

**Comment by PCManticore** [comment]
This works with the master branch of astroid and pylint (1.5.0, respectively 2.0). Please test them if you have a chance and let me know if it works for you.

**Comment by crbunney** [comment]
Confirmed fixed with commit [67f1c8d](https://github.com/PyCQA/astroid/commit/67f1c8de3e07bd3d969c460a78aa6a0e7675626a) of astroid and [64fd450](https://github.com/PyCQA/pylint/commit/64fd4508006b7ceeeb4fc7eac139abfc97a2e116) of pylint
**pylint --version output**
```
No config file found, using default configuration
pylint 2.0.0,
astroid 1.5.0
Python 2.7.11 (default, Jan 22 2016, 08:29:18)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

**pip freeze**
```
astroid==1.5.0
backports.functools-lru-cache==1.2.1
configparser==3.5.0
enum34==1.1.6
isort==4.2.5
lazy-object-proxy==1.2.2
mccabe==0.5.0
pylint==2.0.0
singledispatch==3.4.0.3
six==1.10.0
wrapt==1.10.8
```

Do you know when these versions are likely to be published to PyPi?

**Comment by PCManticore** [comment]
You can check the milestones here: https://github.com/PyCQA/pylint/milestones, although 2.0 might be a bit late (one week or so)

**Comment by crbunney** [comment]
Great, thanks!
